src/erlcloud_cloudformation.erl

```diff
@@ -659,6 +683,113 @@ extract_account_limit(XmlNode) ->
             {value, "Value", optional_text}
         ], XmlNode).
 
-
-
+create_stack_input_to_params(Spec) ->
+    lists:flatten([
+        base_create_stack_input_params(Spec),
+        member_object_array("Parameters", fun cloudformation_parameter_fields/1, Spec#cloudformation_create_stack_input.parameters),
+        member_list("Capabilities", Spec#cloudformation_create_stack_input.capabilities),
+        member_list("NotificationARNs", Spec#cloudformation_create_stack_input.notification_arns),
+        member_list("ResourceTypes", Spec#cloudformation_create_stack_input.resource_types),
+        member_object(
+            "RollbackConfiguration",
+            fun cloudformation_rollback_configuration_fields/1,
+            Spec#cloudformation_create_stack_input.rollback_configuration
+        ),
+        member_object_array("Tags", fun cloudformation_tag_fields/1, Spec#cloudformation_create_stack_input.tags)
+    ]).
+
+base_create_stack_input_params(Spec) ->
+    [
+        { "ClientRequestToken", Spec#cloudformation_create_stack_input.client_request_token },
+        { "DisableRollback", Spec#cloudformation_create_stack_input.disable_rollback },
+        { "EnableTerminationProtection", Spec#cloudformation_create_stack_input.enable_termination_protection },
+        { "OnFailure", Spec#cloudformation_create_stack_input.on_failure },
+        { "RoleARN", Spec#cloudformation_create_stack_input.role_arn },
+        { "StackName", Spec#cloudformation_create_stack_input.stack_name },
+        { "StackPolicyBody", Spec#cloudformation_create_stack_input.stack_policy_body },
+        { "StackPolicyUrl", Spec#cloudformation_create_stack_input.stack_policy_url },
+        { "TemplateBody", Spec#cloudformation_create_stack_input.template_body },
+        { "TemplateURL", Spec#cloudformation_create_stack_input.template_url },
+        { "TimeoutInMinutes", Spec#cloudformation_create_stack_input.timeout_in_minutes }
+    ].
+
+member_object(_Prefix, _Fun, undefined) ->
+    [];
+member_object(Prefix, FieldFun, Object) ->
+    lists:map(
+        fun({Key, Value}) ->
+            {Prefix ++ "." ++ Key, Value}
+        end,
+        FieldFun(Object)
+    ).
+
+member_object_array(Prefix, FieldFun, List) when is_list(List) ->
+    lists:flatten(lists:foldl(
+        fun(Parameter, Acc) ->
+            [lists:map(
+                fun({Key, Value}) ->
+                    {Prefix ++ ".member." ++ integer_to_list(length(Acc)+1) ++ "." ++ Key, Value}
+                end,
+                FieldFun(Parameter)
+            ) | Acc]
+        end,
+        [],
+        List
+    ));
+member_object_array(_Prefix, _Fun, _) ->
+    [].
+
+member_list(Prefix, List) when is_list(List) ->
```

Comment:
we can probably reuse `erlcloud_util:encode_list()`.
And also worth moving the rest of `member...()` function into `erlcloud_util.erl`

Reply:
undefined get filtered out here https://github.com/erlcloud/erlcloud/blob/master/src/erlcloud_aws.erl#L1055 we just need to worry about filtering out nested fields like parameters, capabilities etc